python/gps/algorithm/algorithm_traj_opt.py

```diff
@@ -98,15 +105,107 @@ def _stepadjust(self, m):
 
         self._set_new_mult(predicted_impr, actual_impr, m)
 
+    def _update_cost(self):
+        """ Update the cost objective in each iteration. """
+        # Estimate the importance weights for fusion distributions.
+        itr = self.iteration_count
+        M = len(self.prev)
+        ix = range(self.dX)
+        iu = range(self.dX, self.dX + self.dU)
+        init_samples = 5
+        demo_sum, samples_sum, demoU, demoX = {}, {}, {}, {}
+        for i in xrange(self.M):
+            demoU[i] = self.cur[i].demo_list.get_U()
+            demoX[i] = self.cur[i].demo_list.get_X()
+
+        # itration_count + 1 distributions to evaluate
+        # T: summed over time
+        samples_prob, demos_prob, samples_prob_T, demos_prob_T = {}, {}, {}, {}
+        # number of demo distributions
+        Md = len(demoX)
+        demo_traj = {}
+        # estimate demo distributions empirically
+        for i in xrange(Md):
+            if self._hyperparams['demo_distr_empest']:
+                demo_traj[i] = fit_emp_controllers(demoX[i], demoU[i])
+            # Use LQR estimations?
+
+        for i in xrange(M):
```

Comment:
Generally, this function looks good! You kept the relevant pieces of code. I haven't double-checked the math, but a quick numerical test should verify it.